src/llvm/transforms.jl

```diff
@@ -476,6 +476,68 @@ function memcpy_alloca_to_loadstore(mod::LLVM.Module)
     end
 end
 
+# Split a memcpy into an sret with jlvaluet into individual load/stores
+function memcpy_sret_split!(mod::LLVM.Module)
+    dl = datalayout(mod)
+    ctx = context(mod)
+	    sretkind = LLVM.kind(if LLVM.version().major >= 12
```

Comment:
Is the LLVM version check to be extra safe?  Technically, oldest supported version at this point is 15 (the one coming with Julia v1.10), so you could just inline the check here and below.